experiments/dsl/src/main/scala/org/apache/gearpump/streaming/dsl/Stream.scala

```diff
@@ -127,25 +133,38 @@ class Stream[T](dag: Graph[Op, OpEdge], private val thisNode: Op, private val ed
    * @tparam R
    * @return
    */
-  def process[R](processor: Class[_ <: Task], parallism: Int, description: String = null): Stream[R] = {
+  def process[R: ClassTag](processor: Class[_ <: Task], parallism: Int, description: String = null)(implicit graph: Graph[Op,OpEdge]): Stream[R] = {
     val processorOp = ProcessorOp(processor, parallism, Option(description).getOrElse("process"))
-    dag.addVertex(processorOp)
-    dag.addEdge(thisNode, edge.getOrElse(Shuffle), processorOp)
-    new Stream(dag, processorOp, Some(Shuffle))
+    graph.addVertex(processorOp)
+    graph.addEdge(thisNode, edge.getOrElse(Shuffle), processorOp)
+    new Stream[R](processorOp, Some(Shuffle))
+  }
+
+  /**
+   * connect with a low level Processor(TaskDescription)
+   * @param processors
+   * @return
+   */
+  def branch(processors: ProcessorOp[_ <: Task]*)(implicit graph: Graph[Op,OpEdge]): Stream[T] = {
```

Comment:
Stream.process does not allow branching - only linear downstream chaining. This allows multiple downstream nodes to be attached at a single upstream node.